### PR TITLE
[Serve] Enable multiple cloud in resources

### PR DESCRIPTION
**Pull request by cblmemo**
Originally the SkyServe does not allow multiple resources using different cloud. This PR fixed the issue.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/serve/core.py [hunk 1]
```diff
@@ -60,7 +60,12 @@ def up(
         with ux_utils.print_exception_no_traceback():
             raise RuntimeError('Service section not found.')
 
+    # Pickup the cloud if all resources are from the same cloud. This is to
+    # select the cloud for the controller if it does not exist yet. If the
+    # controller and all replicas are from the same cloud, it should provide
+    # better connectivity.
```
Comment (MaoZiming) [inline sky/serve/core.py:66]:
```suggestion
    # If all resources are from the same cloud, set the cloud of the controller
    # to be that cloud. If the controller and all replicas are from the same cloud,
    # it should provide better connectivity.
```

**Review by MaoZiming — APPROVED** [review]
LG

File: sky/serve/core.py [hunk 2]
```diff
@@ -60,7 +60,12 @@ def up(
...
     requested_cloud: Optional['clouds.Cloud'] = None
+    requested_multiple_clouds = False
```
Comment (MaoZiming) [inline sky/serve/core.py:68]:
```suggestion
    requested_multiple_clouds: bool = False
```

**Comment by Michaelvll** [comment]
If multiple resources is specified for the services, should we let the controller to be launched within those clouds only? We can do something as follows:
- If controller did not exist, make sure the controller is launched within the clouds specified
- If controller already exist, just reuse the controller

**Comment by cblmemo** [comment]
> If multiple resources is specified for the services, should we let the controller to be launched within those clouds only? We can do something as follows:
> 
> * If controller did not exist, make sure the controller is launched within the clouds specified
> * If controller already exist, just reuse the controller

Good point! Fixed in #3231. Thanks!

**Pull request merged**